Fix the parser when using nested output filters

JoshuaLuckers opened the pull request:
### What does it do?
Order the matching results in the order they match and don't include the full matched string. This way to parser doesn't break.

### Why is it needed?
It fixes a problem where nested output filters might break in certain situations.

### Related issue(s)/PR(s)
Fixes issue #14095

JoshuaLuckers commented:
This PR is a work in progress.

Jako commented:
All tests are failing at the moment …

JoshuaLuckers commented:
>All tests are failing at the moment … 

Yes, forgot to make this PR a draft (can’t be changed afterwards).

JoshuaLuckers committed:
Fix the parser when using nested output filters

JoshuaLuckers commented:
@cla-bot check

Jako commented:
I don't get the clue on the change of the regular expression. What do you test with this? It includes the next signs until a colon or a equal sign or the full pattern. This does not sound right to me.

The regular expression that retrieves the content of the surrounding backticks after a modifier has to include everything and can't stop at the first nested backtick of a MODX tag inside. So it 'just' has to ignore the content between all nested MODX tags. I don't know if that's possible in a direct way, since you have to detect opening and closing nested pairs of MODX tags.

JoshuaLuckers commented:
@Jako to see what it does: https://regex101.com/r/N6dGM0/1

You're right, what's going on here isn't right. I added a new chunk of test data to the Unit Test (without this change the test failed) to make sure I didn't break anything. 
Back then I didn't know it would have impact elsewhere and break things (there was also no Unit Test testing the other part).

As you can see in issue #16186 I've now added a Unit Test in the right place, in my local repo I added test data that fails the Unit Test: https://github.com/JoshuaLuckers/revolution/blob/004b877ad0e597e77b19d4a6a09477495bdde4ad/_build/test/Tests/Model/Filters/modInputFilterTest.php

> I don't know if that's possible in a direct way, since you have to detect opening and closing nested pairs of MODX tags.
Correct, I spend several hours to fix it but it's not as easy as it seems.

 I'm in favor of reverting this PR, what do you think @opengeek ?

Jako commented:
If I am right, this issue occurs, when the MODX tags inside of the modifier chain options are not replaced. Normally the parser should have replaced them already. But if they are uncached, this is not the case. Can there be are other reasons?

I am also in favor of reverting this PR.

rthrash commented:
This pull request has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/modx-3-generated-don-t-cache-tags-seem-not-to-work-anymore/5247/21

rthrash commented:
This pull request has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/modx-3-generated-don-t-cache-tags-seem-not-to-work-anymore/5247/23
